contracts/BC_fusion/BSCGovernor.sol

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorSettingsUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/GovernorUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/compatibility/GovernorCompatibilityBravoUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorVotesUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorTimelockControlUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorVotesQuorumFractionUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorPreventLateQuorumUpgradeable.sol";
+
+
+contract BSCGovernor is
+    GovernorUpgradeable,
+    GovernorSettingsUpgradeable,
+    GovernorCompatibilityBravoUpgradeable,
+    GovernorVotesUpgradeable,
+    GovernorTimelockControlUpgradeable,
+    GovernorVotesQuorumFractionUpgradeable,
+    GovernorPreventLateQuorumUpgradeable
+{
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(
+        IVotesUpgradeable _token,
+        TimelockControllerUpgradeable _timelock,
```

Comment:
All the contract should be system contract, we can not depend on a timerlocker that is deployed somewhere else. So please make timerlocker contract one of the system contract.

Reply:
added